test/vstest.console.UnitTests/Processors/CLIRunSettingsArgumentProcessorTests.cs

```diff
@@ -133,6 +134,96 @@ public void InitializeShouldIgnoreWhiteSpaceInBeginningOrEndOfKey()
             Assert.AreEqual(RunSettingsWithDeploymentDisabled, settingsProvider.ActiveRunSettings.SettingsXml);
         }
 
+        [DynamicData(nameof(ValidTestCases), DynamicDataSourceType.Method)]
+        [DataTestMethod]
+        public void InitializeShouldValidateTestRunParameter(string arg, string runSettingsWithTestRunParameters)
+        {
+            var args = new string[] { arg };
+
+            this.executor.Initialize(args);
+
+            Assert.IsNotNull(this.settingsProvider.ActiveRunSettings);
+            Assert.AreEqual(runSettingsWithTestRunParameters, settingsProvider.ActiveRunSettings.SettingsXml);
+        }
+
+        [DynamicData(nameof(InvalidTestCases), DynamicDataSourceType.Method)]
+        [DataTestMethod]
+        public void InitializeShouldThrowErrorIfTestRunParameterNodeIsInValid(string arg)
+        {
+            var args = new string[] { arg };
+            var str = string.Format(CommandLineResources.InvalidTestRunParameterArgument, arg);
+
+            CommandLineException ex = Assert.ThrowsException<CommandLineException>(() => this.executor.Initialize(args));
+
+            Assert.AreEqual(str, ex.Message);
+        }
+
+        [DataRow("Testameter.Parameter(name=\"asf\",value=\"rgq\")")]
+        [DataRow("TestRunParameter.Parameter(name=\"asf\",value=\"rgq\")")]
+        [TestMethod]
+        public void InitializeShouldThrowErrorIfTestRunParameterNodeNameIsInValid(string arg)
+        {
+            var args = new string[] { arg };
+            var str = string.Format(CommandLineResources.MalformedRunSettingsKey);
+
+            CommandLineException ex = Assert.ThrowsException<CommandLineException>(() => this.executor.Initialize(args));
+
+            Assert.AreEqual(str, ex.Message);
+        }
+
+        public static IEnumerable<object[]> InvalidTestCases()
```

Comment:
Rename to make it specific to TestRunParameters
Also add this to a region called TestRunParamtersTests